actionpack/CHANGELOG.md

```diff
@@ -1,3 +1,42 @@
+*   Add the `direct` method to the routing DSL
+
+    This new method allows customization of the routing behavior in two ways:
+
+    1.  Custom url helpers:
+
+        ``` ruby
+        direct(:apple) { "http://www.apple.com" }
+
+        >> apple_url
+        => "http://www.apple.com"
+        ```
+
+        This has the advantage of being available everywhere url helpers are available
+        unlike custom url helpers defined in helper modules, etc.
+
+    2.  Custom polymorphic mappings:
+
+        ``` ruby
+        resource :basket
+        direct(class: "Basket") { [:basket] }
+        ```
+
+        ``` erb
+        <%= form_for @basket do |form| %>
+        <!-- basket form -->
+        <% end %>
+        ```
+
+        This generates the correct singular URL for the form instead of the default
+        resources member url, e.g. `/basket` vs. `/basket/:id`.
```

Comment:
IMO we should come up with a different example here -- something that uses a parent route, perhaps -- both because it's desirable to show an argument-using block, and because I think `resource` should be registering the singular URL automatically (with some conflict resolution to avoid an incompatible change if you define both singular and plural resources for the same class).